{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This workflow uses blast to match symbiosis genes from other studies (Rodriguez-Lanetty et al. 2006, Lehnert et al. 2013, and Li et al. 2018) to the _A. elegantissima_ transcriptome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## First download _A. elegantissima_ transcriptome from Meyer Lab website and turn into blast database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/jd/Documents/Projects/Anthopleura/Methylation-splicing/Alt-splicing/data\n"
     ]
    }
   ],
   "source": [
    "cd ../data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gunzip: can't stat: Aele_transcriptome_v1.fasta.gz (Aele_transcriptome_v1.fasta.gz.gz): No such file or directory\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 40.7M  100 40.7M    0     0  6711k      0  0:00:06  0:00:06 --:--:-- 7898k\n"
     ]
    }
   ],
   "source": [
    "!curl -O http://meyerlab:coral@files.cgrb.oregonstate.edu/Meyer_Lab/transcriptomes/Aele/Aele_transcriptome_v1.fasta.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!gunzip Aele_transcriptome_v1.fasta.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 12/09/2018 18:19:03\n",
      "New DB name:   /Users/jd/Documents/Projects/Anthopleura/Methylation-splicing/Alt-splicing/data/Aele_transcriptome_v1.fasta\n",
      "New DB title:  Aele_transcriptome_v1.fasta\n",
      "Sequence type: Nucleotide\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 142934 sequences in 4.97578 seconds.\n"
     ]
    }
   ],
   "source": [
    "#Make blast database out of transcriptome\n",
    "!makeblastdb -in Aele_transcriptome_v1.fasta -parse_seqids -dbtype nucl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now obtain symbiosis-related genes from earlier studies on A. elegantissima and Aiptasia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract symbiosis genes from [Rodriguez-Lanetty et al. 2006](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-7-23). This was done by copying accession numbers from the manuscript and searching NCBI. This produced the following FASTA file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">lcl|DQ309521.1_cds_ABD04163.1_1 [protein=plasma glutamate carboxypeptidase-like protein] [frame=2] [protein_id=ABD04163.1] [location=<1..>712] [gbkey=CDS]\r\n",
      "TTTACTGATACATTTGGCAATAGAATATCAGGGTCGAAGAATTTAGAGAATTCCATTGATTACATGTTAG\r\n",
      "AAAAATTACATAAAGATGGGCTGGAGAATGTTCATGGAGAAGAAGTGAAAGTGCCGCACTGGGTTAGAGG\r\n",
      "AAACGAGTCAGCCACTATGCTAAGCCCGAGAAAATATTCCATGAATATCTTAGGGCTGGGATCAAGCGTT\r\n",
      "GGAACCCCGCCAGAAGGGATCACTGCAGATGTCTTGGTTGTTAAATCATTTGATGAGCTAAAAACCAGGG\r\n",
      "CGAAAGAGGCCAAAGGAAAGATAGTTGTCTTCAATGAAGACTTCGTATCGTATGGAGAAACTGTTTCTTA\r\n",
      "TCGGGATTTTTCCGCGGTTCACACGGCTGAGGTTGGTGGTGTTGCAACTCTTGTGAGGTCCGTGGCTTCA\r\n",
      "TTTTCGATCAACAGTCCACACACTGGATGGCAGGACTATAGTAAAGGTGTAAAGAAAATCCCAACTGCCT\r\n",
      "GTATTACAATCGAAGATGCTGAGATGTTGGGGAGAATGGCGGCCCGAGGAACCAAGATTACGATCAACCT\r\n",
      "AAAAATGGAAGCCAAGACTCTCCCTCCTGTTACGTCCAGAAACACGGTGGCTGAGATTGTTGGCAGTAAA\r\n"
     ]
    }
   ],
   "source": [
    "!head Rodriguez_Lanetty_seqs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Blast search transcriptome for these genes, send output to ../analyses directory\n",
    "!blastn -query Rodriguez_Lanetty_seqs.fasta -db Aele_transcriptome_v1.fasta -outfmt 6 -evalue 1E-05 -num_threads 8 -out ../analyses/Aele_blastn_Rodriguez-Lanetty.tab "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### Next, symbiosis-related genes were obtained from two studies on _Aiptasia_, [Lehnert et al. 2014](http://www.g3journal.org/content/4/2/277.short) and [Li et al. 2018](http://advances.sciencemag.org/content/4/8/eaat2142). Protein sequences (symbio_prot.fasta) were obtained from uniprot ids (symbiosis_uniprot.txt)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NP_003807.1 disintegrin and metalloproteinase domain-containing protein 9 precursor [Homo sapiens]\r\n",
      "MGSGARFPSGTLRVRWLLLLGLVGPVLGAARPGFQQTSHLSSYEIITPWRLTRERREAPRPYSKQVSYVI\r\n",
      "QAEGKEHIIHLERNKDLLPEDFVVYTYNKEGTLITDHPNIQNHCHYRGYVEGVHNSSIALSDCFGLRGLL\r\n",
      "HLENASYGIEPLQNSSHFEHIIYRMDDVYKEPLKCGVSNKDIEKETAKDEEEEPPSMTQLLRRRRAVLPQ\r\n",
      "TRYVELFIVVDKERYDMMGRNQTAVREEMILLANYLDSMYIMLNIRIVLVGLEIWTNGNLINIVGGAGDV\r\n",
      "LGNFVQWREKFLITRRRHDSAQLVLKKGFGGTAGMAFVGTVCSRSHAGGINVFGQITVETFASIVAHELG\r\n",
      "HNLGMNHDDGRDCSCGAKSCIMNSGASGSRNFSSCSAEDFEKLTLNKGGNCLLNIPKPDEAYSAPSCGNK\r\n",
      "LVDAGEECDCGTPKECELDPCCEGSTCKLKSFAECAYGDCCKDCRFLPGGTLCRGKTSECDVPEYCNGSS\r\n",
      "QFCQPDVFIQNGYPCQNNKAYCYNGMCQYYDAQCQVIFGSKAKAAPKDCFIEVNSKGDRFGNCGFSGNEY\r\n",
      "KKCATGNALCGKLQCENVQEIPVFGIVPAIIQTPSRGTKCWGVDFQLGSDVPDPGMVNEGTKCGAGKICR\r\n"
     ]
    }
   ],
   "source": [
    "!head symbio_prot.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#blast protein sequences against A. elegantissima transcriptome to find homologs\n",
    "!tblastn -query symbio_prot.fasta -db Aele_transcriptome_v1.fasta -outfmt 6 -evalue 1E-05 -num_threads 8 -out ../analyses/Aele_tblastn_symbio.tab "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NP_003807.1\tcomp17218_c0_seq1\t40.179\t448\t216\t17\t347\t768\t61\t1326\t3.57e-75\t266\r\n",
      "NP_003807.1\tcomp40514_c0_seq1\t34.579\t428\t261\t11\t91\t508\t1\t1257\t5.54e-72\t253\r\n",
      "NP_003807.1\tcomp40514_c0_seq2\t34.272\t426\t261\t11\t93\t508\t2462\t1212\t3.41e-70\t248\r\n",
      "NP_003807.1\tcomp13383_c0_seq5\t33.408\t449\t271\t11\t78\t509\t213\t1526\t2.16e-68\t248\r\n",
      "NP_003807.1\tcomp13383_c0_seq5\t32.530\t166\t105\t4\t510\t670\t2403\t2894\t2.75e-18\t92.0\r\n",
      "NP_003807.1\tcomp13383_c0_seq4\t33.408\t449\t271\t11\t78\t509\t213\t1526\t2.39e-68\t248\r\n",
      "NP_003807.1\tcomp13383_c0_seq4\t32.530\t166\t105\t4\t510\t670\t2403\t2894\t2.79e-18\t92.0\r\n",
      "NP_003807.1\tcomp40514_c0_seq3\t35.149\t404\t244\t10\t114\t508\t26\t1210\t2.51e-68\t243\r\n",
      "NP_003807.1\tcomp40514_c0_seq4\t35.149\t404\t244\t10\t114\t508\t26\t1210\t2.51e-68\t243\r\n",
      "NP_003807.1\tcomp13383_c0_seq3\t33.408\t449\t271\t11\t78\t509\t213\t1526\t2.93e-68\t248\r\n"
     ]
    }
   ],
   "source": [
    "!head ../analyses/Aele_tblastn_symbio.tab "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#make sure blast hits are below e-05\n",
    "!awk '$11<0.000005{print $0}' Aele_tblastn_transcriptome_cat_symbio.tab > Aele_tblastn_transcriptome_cat_symbio_eval.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#get unique transcriptome records (column2)\n",
    "!sort -uk2,2 Aele_tblastn_transcriptome_cat_symbio_eval.tab > Aele_tblastn_transcriptome_cat_symbio_eval_unique.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Just get seq ids\n",
    "!awk '{print $2}' Aele_tblastn_transcriptome_cat_symbio_eval_unique.tab > Aele_tblastn_transcriptome_cat_symbio_seqs.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    9931    9931  180281 Aele_tblastn_transcriptome_cat_symbio_seqs.tab\r\n"
     ]
    }
   ],
   "source": [
    "!wc Aele_tblastn_transcriptome_cat_symbio_seqs.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#extract seqs of interest to new file\n",
    "!faSomeRecords ./Anthopleura_genome/Aele_transcriptome_v1.fasta Aele_tblastn_transcriptome_cat_symbio_seqs.tab sym_seqs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9931\r\n"
     ]
    }
   ],
   "source": [
    "!grep -c \"^>\" sym_seqs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 1.80 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=479864332, availableWord=45764544\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 74507308 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 138591820 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 195546316 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 246163596 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 291148348 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 331126876 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 366655836 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 398230076 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 426289292 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 451224348 characters processed.\n",
      "[BWTIncConstructFromPacked] 110 iterations done. 473382668 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 114 iterations.\n",
      "[bwa_index] 147.21 seconds elapse.\n",
      "[bwa_index] Update BWT... 1.51 sec\n",
      "[bwa_index] Pack forward-only FASTA... 1.05 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 56.93 sec\n",
      "[main] Version: 0.7.16a-r1181\n",
      "[main] CMD: /Users/jd/bwa-0.7.16a/bwa index ./Anthopleura_genome/wtdbg2_genome/Ae.ctg.lay.fa\n",
      "[main] Real time: 210.733 sec; CPU: 208.516 sec\n"
     ]
    }
   ],
   "source": [
    "#index draft genome\n",
    "!/Users/jd/bwa-0.7.16a/bwa index ./Anthopleura_genome/wtdbg2_genome/Ae.ctg.lay.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 3394 sequences (10002491 bp)...\n",
      "[M::process] read 4292 sequences (10002175 bp)...\n",
      "[M::mem_process_seqs] Processed 3394 reads in 40.952 CPU sec, 42.840 real sec\n",
      "[M::process] read 2245 sequences (2216747 bp)...\n",
      "[M::mem_process_seqs] Processed 4292 reads in 36.086 CPU sec, 38.440 real sec\n",
      "[M::mem_process_seqs] Processed 2245 reads in 6.697 CPU sec, 7.079 real sec\n",
      "[main] Version: 0.7.16a-r1181\n",
      "[main] CMD: /Users/jd/bwa-0.7.16a/bwa mem ./Anthopleura_genome/wtdbg2_genome/Ae.ctg.lay.fa sym_seqs.fasta\n",
      "[main] Real time: 88.819 sec; CPU: 84.155 sec\n"
     ]
    }
   ],
   "source": [
    "#map transcriptome sequences to draft genome\n",
    "!/Users/jd/bwa-0.7.16a/bwa mem ./Anthopleura_genome/wtdbg2_genome/Ae.ctg.lay.fa sym_seqs.fasta > sym_seqs_map.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view -Sb sym_seqs_map.sam > sym_seqs_map.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools sort sym_seqs_map.bam -o sym_seqs_map.sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools index sym_seqs_map.sorted.bam "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bedtools bamtobed -i sym_seqs_map.sorted.bam > sym_seqs.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Get unique mapped seqs\n",
    "!sort -u -k4,4 sym_seqs.bed > sym_seqs_uniq.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    9760   58560  412071 sym_seqs_uniq.bed\r\n"
     ]
    }
   ],
   "source": [
    "#793 sequences out of 803 were mapped\n",
    "!wc sym_seqs_uniq.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Get number of unique contigs that had mappings\n",
    "!sort -u -k1,1 sym_seqs.bed > sym_seqs_uniq_contig.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    1724   10344   71543 sym_seqs_uniq_contig.bed\r\n"
     ]
    }
   ],
   "source": [
    "#number of unique contigs that had mappings\n",
    "!wc sym_seqs_uniq_contig.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5359\r\n"
     ]
    }
   ],
   "source": [
    "#total number of contigs\n",
    "!grep -c \"^>\" ./Anthopleura_genome/wtdbg2_genome/Ae.ctg.lay.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!bedtools sort -i sym_seqs_uniq.bed > sym_seqs_uniq.sorted.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#merge mappings withing distance of 2000 bp. Average intron length is 3,479 according to Hong et al. (2006)\n",
    "!bedtools merge -d 2000 -i sym_seqs_uniq.sorted.bed > sym_seqs_merged.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat ./pass/barcode03/*.fastq > cat.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!awk '{if(NR%4==1) {printf(\">%s\\n\",substr($0,2));} else if(NR%4==2) print;}' cat.fastq > barcode03.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">23fe984c-edc1-493f-9929-f9c49254f867 runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=173 ch=233 start_time=2018-10-19T02:04:53Z barcode=barcode03\r\n",
      "TTCGGTATTGCTTCGTTCCGGTTTATCAGGTGATTGCTGCTGCCATACCCTTGCTGCCAACGTTGCGTTGTTGATTACCGTTGCTGCGGTTATGCTGCTGCCGTTTCTGATTACCTGCCGCCTGCCGTTTTGTTCTGGATGTTTAACGGTCGCCGCCCTACGTGACGAGTCTTGTGTCCCAGTTACCAGGACACCTGTCGCTCTATCTTCTTTTTTTTTTTTTTTTTGCGCGCTTAATATCCCAGCACCAACGAAACCTGTTAGCTGGGACACAAGACTCGTCACACCGCA\r\n",
      ">44b3c6d4-5134-4e93-a03a-aa9d28862023 runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=209 ch=433 start_time=2018-10-19T02:05:41Z barcode=barcode03\r\n",
      "TCGAAGCGTACTTCGTTCGGCAGTTCAGATTGGGTGTTTAACCGGATAATAAAGCCTACCGTGACGAGTCTTGTGTCCCAGTTACCAGGACTTGCCGCTGTCGCTCTATCTTCTTTTTTACTTGGAAATAAATGCTTGTTCAGTAGATATGTATTAGACATTAAAGGTAAAGGATTATTCTTGAGTATATCTGTAGGAAAACTTTTGTCTACGCCAGCATGGAAGGACCAACCACTTTTCATCCGAATCTTCATATGCCATCGCTTTATGCGCCACAAATACTATGCATCATATTACGTACAATTTTGATAAATAAAATAAAATAATCTCTGTGTTTTGGAAACTTATCTATTCAATAATTTTAATGATCAAGAATTTCAGCTTACTGGTCATTAAAATTCGTAAGGTCCGCAAGGAACAACTTATTTAAATTCACATATCTGTATCAAAAAGGTCAAATATTTTCGCTTTTCTTCCTTTCATTTACAAATAAAAATAAATAAATGTACTCTGTTCAAATGTTCAGTTTATACCGTAGTTTTCTGTATTTTGGCTGTGACTTATTGCGGTCAGAGGCCACCGTCGTATCGGTTGCATTCAATACAGTGGATAAGCGTTCGCATGAAGCTGATGCGATAACATCATCATATTTCGTTGGAAATAATTTCAATTTAACAATATCAAGCACTGCTTTCGGCTCCTGAAAAGTAACTGGAACCTGAAAGACTCAAATCGTA\r\n",
      ">2fe84a61-5656-4c43-b9c1-f27a95ec3f7f runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=1490 ch=19 start_time=2018-10-19T02:14:56Z barcode=barcode03\r\n",
      "ATCGTACTTCGTTCAGTTCGTTGGGTGTTATGGATCATCATCGCGACCGATCTTGTGTCCCAGTTACCAGGTTTCTGTTAGTTGCTGATATTGCTGGAGCAAAATTGACAAAGATATTATACATAGACCCTTACAGCACGTCGGAATGCAGCTCAATTTGGAGGATCGCTCATGGGAATTCTGACGTATAAGTCATTCTGTTCATGAAAGGGTCCTCTCCCCATACTCTATGTCATTTCCATAAGATTTAACCTTAGCATTCTATATAGATGTCAGGTCTTTGAAAAAAGATAGAGCGACAGACGGAGTCCTGGTAACTGGGACCTAAGACTCGTCACGCA\r\n",
      ">2e89a3df-f2d6-4952-97e1-4c354ca60b5b runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=2048 ch=29 start_time=2018-10-19T02:25:04Z barcode=barcode03\r\n",
      "TTCGTTTTTTTTTTTTTTTCCATTCCTTCATTCCATTCCAGATCATCCCAGTTACCAGGACTTCCTGTCAGCTCTATCTTCTTTTTTTTTTGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTGGACTTCATTCCTAAGTTCGTCCATTCCTGGTCACCACTCGTTCCTGGGCTTCCATGGAACTGTTTCCATGATTTGTATCACGTGACAGTTTCTTCTCACTCACCCAGCAATGTCGGCACCAGCTTGAGCCTGGTGGCTGGGACACAAGAGACTCGTCCTC\r\n",
      ">970120d7-1a8f-413d-8cde-db8b94a6ecb7 runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=56605 ch=103 start_time=2018-10-19T13:24:14Z barcode=barcode03\r\n",
      "TTGGTAATACTTCAGTTCAGTTCGGTGGGTGTTTAACCGGATAAAATATACAATGACGAGTCACCCAGTAATCCCAGTTACACCAGGTTTCTGTTAGTGCTGATATTGCTGGGGATTCCAGTTGATGGATGGAGATGGTGGGCGTTCGCGTATAAACGGCAGTTCCGCTTCCTGGATTTTGATACAATTTTATGGGTGTCGCTTCTGTTTTGTATAAAAATAAATATTAATACATATATTTCTTAAAAAGATAGAGCGACAGACAGGTCTGGTAACTGGGACACAAGACTCGTGGCA\r\n"
     ]
    }
   ],
   "source": [
    "!head barcode03.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat ./pass/barcode04/*.fastq > cat04.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!awk '{if(NR%4==1) {printf(\">%s\\n\",substr($0,2));} else if(NR%4==2) print;}' cat04.fastq > barcode04.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">e5d65991-83be-43ac-a2a5-24d97e67b9fe runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=64444 ch=78 start_time=2018-10-19T13:22:15Z barcode=barcode04\r\n",
      "TTAGAGTATTGCTTCGTTCCGATCAGGTGATTGTTTAACCGGATGCATCATCTACCGTGACTTCGGATTCTATCGGTGTTTCCTAACTTGCCTGTCCGCTCTATCTTCTTTTTTAGATTAAAGAGCGTTTTCCCATTGGTGATTACAGCGGGTCCTATCTCTTACTTACCGACACACAGACCTTCTGGCTGGACACTTGTTTGAGCGGCCTTCTGTTTACCTGCCACTTACACCTGGAGCTGTTGCCTTGGTCGCCTTTTAATTTGTTCCTTTGCTCCTGCTTCTCTTTACTTGAAGCCCCACCGGTGGCTTCCAGTCTATTTATTGCTTTTCTCACTTCTGGCTTCAGTTTCTCGCTTTTGCTTAATATTTTCCATTGTTGCCGCACCCTGAATGTGCCAGATCTCTGAGCGCCAGGCGATTACGGCGGAGTTCTCTTCTTGGACCGCCTCTTCTTGTGTTCCTTTCTTGTGCTTACGGCCGGTAAAAGCACAGGATCCGGAGGGGTTGACGCATGAAGTGCATTGGCTCCTTTTTGTTTAGAAGAGTTAAATGCACCCGCCATCAAGACGGACGTATCTTTTACTGCCGTGGCGCAGGTGGATTTTATTACCGCCGCTGTTGCAAAGCTCGGGCTAAATGGCAGAGGATGTTTCGGATCCGAAGAGCCCCAGCAATATCCAGCACCAAAACGGAAATAAGAACACGATGGAATCAGGGAGTCACGCA\r\n",
      ">784dd2f9-4049-4b86-88f4-95bdef9aa6be runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=71860 ch=88 start_time=2018-10-19T13:27:50Z barcode=barcode04\r\n",
      "TGGTATGCTTCGTTCAGTTTATCAGGCCAGGTGTTCCGCCAGACCGCCACGCTCCACCGTGACTTCCGGATTATCGTGTTTCCCTATTTCGTGGTGTGATATTGCTGGGTCGGGAAGTCGTCACAATGACCACGAAATGACGTCATACTCTCAACTTTGATTTTTGATACAAAGTATAACCTGTTCCTACTGGTTTTACTGACCTGTTATTTTACACAGGTTATCTATTACGGTCGAGATTCCTGGGGTAATTCAGTCATACAAATATTAAGGTGGCAGGATGTACTTTATCTGTTTCCATAGCAACAAGGTATGCTTCACTGTAATGCTACATAGGCGTACGCAGTGATGAACGCCAAAACGATAAAAACTCAGTAATGAGAAGTCTTGAAATGCGCTAAACGAAGCTATGGTGTTGTACCATCAATAAAATTAAAGTCAGTTCCTGGCAGCTTTTAATTTGTGCTGGATCCAGAATCGCAATGAAATCCGCAATCCATAGAAAATAAGCTTAATATTGTTGTTGCATAACTCACATGAACCTGAACAGTTTACAATAAATATCCTATTATATTTAGTATGTAGCCAGTAATAAAAACAAAAAGATAGAACGGCGAGCAGGCAGGATTAAGAACGCGATAGAATCCAGGGGAATCACG\r\n",
      ">72f67dfa-ed00-4811-811a-dbf5b238478b runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=64375 ch=422 start_time=2018-10-19T13:48:14Z barcode=barcode04\r\n",
      "TGGGTGCCGGTATTGCTGGTTCATTTCCTTGATCGCCTACCGTGACTTCCGGATTATCGTGTTTCCCTATTTCTGTTGGTGCGTATTGCTGGGGAAAAAAGAGATAGAGGCGACAGGCAGCAGGATTGGGGAAGCACATTAGAATCCGGTCACCG\r\n",
      ">3f752c14-d432-4d3e-9563-a10a7fb00f50 runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=68344 ch=374 start_time=2018-10-19T13:55:47Z barcode=barcode04\r\n",
      "TTACAGGTAGCGTTCAGTTCGTTTGGGTGTTTTACTTGATTCGCGCTCGCCTACCGTGACTTCAGTTCTATCGTGTTTCCCTAACGCCTCTGCAGCTCTATCTTCTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTCTTTTTTCCCCAGCAATATCAGCACCAACAGAAATAGGAAGCGATAGAATTCAAAACTCCTC\r\n",
      ">8d05ecae-cd9a-460b-ade2-b466ba4576c5 runid=8fb409436e791ef96e1a9a22b1234bbf145825a9 read=50921 ch=293 start_time=2018-10-19T14:14:47Z barcode=barcode04\r\n",
      "TCCCTCGTTTGCCGGTATTGCGCCTTCACGTTCAATCAGGTGGGTGTTTATGGGATACATCGCTCCCGTGACTTCGGATTCTATCGTGTTTCTATTTCTGTTGGTGCTGATATTGCTGGGCATCATAATGGCTGGAAATAATAAAGTGAAGTTCTTCAACCTCGTTTTATCCTCTGAAATGTTTTAGAATAGATCATGGTCAAATGGGATCGCTTCATCGCTTCACATAACTCAAAACAATAAAAATCAAACGGTTTCGACCAGTTAAAGGCAATGGAATCAGAATCACG\r\n"
     ]
    }
   ],
   "source": [
    "!head barcode04.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Get list of unique genomce contigs that had mappings\n",
    "!awk '{print $1}' sym_seqs_uniq_contig.bed > unique.contigs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#extract seqs of interest to new file\n",
    "!faSomeRecords ./Anthopleura_genome/wtdbg2_genome/Ae.ctg.lay.fa unique.contigs genome_subset.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 1.22 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=294623686, availableWord=32730584\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 53990726 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 99743846 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 140405318 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 176541302 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 208655046 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 237193782 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 262555062 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 285092182 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 85 iterations.\n",
      "[bwa_index] 89.04 seconds elapse.\n",
      "[bwa_index] Update BWT... 1.04 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.76 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 34.42 sec\n",
      "[main] Version: 0.7.16a-r1181\n",
      "[main] CMD: /Users/jd/bwa-0.7.16a/bwa index genome_subset.fasta\n",
      "[main] Real time: 130.674 sec; CPU: 126.495 sec\n"
     ]
    }
   ],
   "source": [
    "#index genome subset \n",
    "!/Users/jd/bwa-0.7.16a/bwa index genome_subset.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 19026 sequences (10000476 bp)...\n",
      "[M::process] read 21178 sequences (10000140 bp)...\n",
      "[M::mem_process_seqs] Processed 19026 reads in 23.300 CPU sec, 23.834 real sec\n",
      "[M::process] read 20866 sequences (10000439 bp)...\n",
      "[M::mem_process_seqs] Processed 21178 reads in 21.415 CPU sec, 21.701 real sec\n",
      "[M::process] read 21232 sequences (10000832 bp)...\n",
      "[M::mem_process_seqs] Processed 20866 reads in 22.801 CPU sec, 22.883 real sec\n",
      "[M::process] read 20606 sequences (10000616 bp)...\n",
      "[M::mem_process_seqs] Processed 21232 reads in 20.926 CPU sec, 21.084 real sec\n",
      "[M::process] read 19804 sequences (10000779 bp)...\n",
      "[M::mem_process_seqs] Processed 20606 reads in 21.498 CPU sec, 21.842 real sec\n",
      "[M::process] read 20878 sequences (10000576 bp)...\n",
      "[M::mem_process_seqs] Processed 19804 reads in 23.186 CPU sec, 23.602 real sec\n",
      "[M::process] read 19104 sequences (10000233 bp)...\n",
      "[M::mem_process_seqs] Processed 20878 reads in 20.760 CPU sec, 20.847 real sec\n",
      "[M::process] read 19430 sequences (10000036 bp)...\n",
      "[M::mem_process_seqs] Processed 19104 reads in 24.013 CPU sec, 26.516 real sec\n",
      "[M::process] read 20006 sequences (10000517 bp)...\n",
      "[M::mem_process_seqs] Processed 19430 reads in 22.770 CPU sec, 24.865 real sec\n",
      "[M::process] read 19430 sequences (10001175 bp)...\n",
      "[M::mem_process_seqs] Processed 20006 reads in 25.169 CPU sec, 30.792 real sec\n",
      "[M::process] read 18608 sequences (10000926 bp)...\n",
      "[M::mem_process_seqs] Processed 19430 reads in 25.977 CPU sec, 28.203 real sec\n",
      "[M::process] read 18848 sequences (10000172 bp)...\n",
      "[M::mem_process_seqs] Processed 18608 reads in 23.044 CPU sec, 23.461 real sec\n",
      "[M::process] read 18978 sequences (10000549 bp)...\n",
      "[M::mem_process_seqs] Processed 18848 reads in 22.953 CPU sec, 23.122 real sec\n",
      "[M::process] read 18456 sequences (10001385 bp)...\n",
      "[M::mem_process_seqs] Processed 18978 reads in 24.910 CPU sec, 25.544 real sec\n",
      "[M::process] read 17808 sequences (10001072 bp)...\n",
      "[M::mem_process_seqs] Processed 18456 reads in 21.280 CPU sec, 21.354 real sec\n",
      "[M::process] read 19132 sequences (10000492 bp)...\n",
      "[M::mem_process_seqs] Processed 17808 reads in 22.467 CPU sec, 22.642 real sec\n",
      "[M::process] read 18082 sequences (10001438 bp)...\n",
      "[M::mem_process_seqs] Processed 19132 reads in 21.185 CPU sec, 21.243 real sec\n",
      "[M::process] read 18108 sequences (10000428 bp)...\n",
      "[M::mem_process_seqs] Processed 18082 reads in 24.319 CPU sec, 25.171 real sec\n",
      "[M::process] read 17200 sequences (10000441 bp)...\n",
      "[M::mem_process_seqs] Processed 18108 reads in 23.170 CPU sec, 23.715 real sec\n",
      "[M::process] read 17044 sequences (10000683 bp)...\n",
      "[M::mem_process_seqs] Processed 17200 reads in 23.454 CPU sec, 23.941 real sec\n",
      "[M::process] read 17894 sequences (10000020 bp)...\n",
      "[M::mem_process_seqs] Processed 17044 reads in 22.868 CPU sec, 23.040 real sec\n",
      "[M::process] read 16964 sequences (10000143 bp)...\n",
      "[M::mem_process_seqs] Processed 17894 reads in 22.653 CPU sec, 22.886 real sec\n",
      "[M::process] read 18250 sequences (10001485 bp)...\n",
      "[M::mem_process_seqs] Processed 16964 reads in 23.119 CPU sec, 23.356 real sec\n",
      "[M::process] read 16560 sequences (10000440 bp)...\n",
      "[M::mem_process_seqs] Processed 18250 reads in 22.642 CPU sec, 22.895 real sec\n",
      "[M::process] read 16516 sequences (10001115 bp)...\n",
      "[M::mem_process_seqs] Processed 16560 reads in 24.035 CPU sec, 24.163 real sec\n",
      "[M::process] read 17092 sequences (10000124 bp)...\n",
      "[M::mem_process_seqs] Processed 16516 reads in 23.175 CPU sec, 23.258 real sec\n",
      "[M::process] read 16482 sequences (10000508 bp)...\n",
      "[M::mem_process_seqs] Processed 17092 reads in 22.133 CPU sec, 22.303 real sec\n",
      "[M::process] read 16584 sequences (10000742 bp)...\n",
      "[M::mem_process_seqs] Processed 16482 reads in 23.027 CPU sec, 23.103 real sec\n",
      "[M::process] read 17146 sequences (10000891 bp)...\n",
      "[M::mem_process_seqs] Processed 16584 reads in 22.502 CPU sec, 22.651 real sec\n",
      "[M::process] read 16466 sequences (10000121 bp)...\n",
      "[M::mem_process_seqs] Processed 17146 reads in 22.480 CPU sec, 22.580 real sec\n",
      "[M::process] read 17266 sequences (10000227 bp)...\n",
      "[M::mem_process_seqs] Processed 16466 reads in 24.295 CPU sec, 24.484 real sec\n",
      "[M::process] read 16462 sequences (10000724 bp)...\n",
      "[M::mem_process_seqs] Processed 17266 reads in 24.529 CPU sec, 24.893 real sec\n",
      "[M::process] read 16098 sequences (10000041 bp)...\n",
      "[M::mem_process_seqs] Processed 16462 reads in 24.126 CPU sec, 24.523 real sec\n",
      "[M::process] read 17446 sequences (10000394 bp)...\n",
      "[M::mem_process_seqs] Processed 16098 reads in 23.926 CPU sec, 24.070 real sec\n",
      "[M::process] read 16202 sequences (10001602 bp)...\n",
      "[M::mem_process_seqs] Processed 17446 reads in 22.563 CPU sec, 22.785 real sec\n",
      "[M::process] read 16672 sequences (10000146 bp)...\n",
      "[M::mem_process_seqs] Processed 16202 reads in 24.136 CPU sec, 24.410 real sec\n",
      "[M::process] read 16714 sequences (10000842 bp)...\n",
      "[M::mem_process_seqs] Processed 16672 reads in 23.212 CPU sec, 23.376 real sec\n",
      "[M::process] read 16036 sequences (10000219 bp)...\n",
      "[M::mem_process_seqs] Processed 16714 reads in 25.187 CPU sec, 25.555 real sec\n",
      "[M::process] read 17506 sequences (10000842 bp)...\n",
      "[M::mem_process_seqs] Processed 16036 reads in 24.933 CPU sec, 25.229 real sec\n",
      "[M::process] read 15970 sequences (10000613 bp)...\n",
      "[M::mem_process_seqs] Processed 17506 reads in 23.231 CPU sec, 23.556 real sec\n",
      "[M::process] read 16198 sequences (10000362 bp)...\n",
      "[M::mem_process_seqs] Processed 15970 reads in 24.408 CPU sec, 25.120 real sec\n",
      "[M::process] read 18212 sequences (10000390 bp)...\n",
      "[M::mem_process_seqs] Processed 16198 reads in 24.324 CPU sec, 24.548 real sec\n",
      "[M::process] read 18452 sequences (10000505 bp)...\n",
      "[M::mem_process_seqs] Processed 18212 reads in 22.408 CPU sec, 22.640 real sec\n",
      "[M::process] read 18264 sequences (10000679 bp)...\n",
      "[M::mem_process_seqs] Processed 18452 reads in 22.551 CPU sec, 23.165 real sec\n",
      "[M::process] read 18806 sequences (10000955 bp)...\n",
      "[M::mem_process_seqs] Processed 18264 reads in 23.869 CPU sec, 24.214 real sec\n",
      "[M::process] read 20180 sequences (10000527 bp)...\n",
      "[M::mem_process_seqs] Processed 18806 reads in 22.749 CPU sec, 23.083 real sec\n",
      "[M::process] read 19808 sequences (10000543 bp)...\n",
      "[M::mem_process_seqs] Processed 20180 reads in 23.743 CPU sec, 24.285 real sec\n",
      "[M::process] read 19738 sequences (10000799 bp)...\n",
      "[M::mem_process_seqs] Processed 19808 reads in 21.702 CPU sec, 21.924 real sec\n",
      "[M::process] read 21162 sequences (10000229 bp)...\n",
      "[M::mem_process_seqs] Processed 19738 reads in 24.708 CPU sec, 25.590 real sec\n",
      "[M::process] read 21938 sequences (10000560 bp)...\n",
      "[M::mem_process_seqs] Processed 21162 reads in 20.162 CPU sec, 20.220 real sec\n",
      "[M::process] read 21992 sequences (10001184 bp)...\n",
      "[M::mem_process_seqs] Processed 21938 reads in 20.273 CPU sec, 20.524 real sec\n",
      "[M::process] read 21808 sequences (10000842 bp)...\n",
      "[M::mem_process_seqs] Processed 21992 reads in 19.870 CPU sec, 19.903 real sec\n",
      "[M::process] read 22416 sequences (10000885 bp)...\n",
      "[M::mem_process_seqs] Processed 21808 reads in 20.688 CPU sec, 20.781 real sec\n",
      "[M::process] read 22116 sequences (10000541 bp)...\n",
      "[M::mem_process_seqs] Processed 22416 reads in 19.695 CPU sec, 19.770 real sec\n",
      "[M::process] read 23180 sequences (10000829 bp)...\n",
      "[M::mem_process_seqs] Processed 22116 reads in 23.464 CPU sec, 23.703 real sec\n",
      "[M::process] read 22354 sequences (10000092 bp)...\n",
      "[M::mem_process_seqs] Processed 23180 reads in 20.009 CPU sec, 20.168 real sec\n",
      "[M::process] read 21672 sequences (10000456 bp)...\n",
      "[M::mem_process_seqs] Processed 22354 reads in 19.899 CPU sec, 19.942 real sec\n",
      "[M::process] read 21796 sequences (10001085 bp)...\n",
      "[M::mem_process_seqs] Processed 21672 reads in 19.317 CPU sec, 19.324 real sec\n",
      "[M::process] read 22402 sequences (10000163 bp)...\n",
      "[M::mem_process_seqs] Processed 21796 reads in 22.388 CPU sec, 22.833 real sec\n",
      "[M::process] read 22130 sequences (10001056 bp)...\n",
      "[M::mem_process_seqs] Processed 22402 reads in 21.057 CPU sec, 21.422 real sec\n",
      "[M::process] read 22202 sequences (10000773 bp)...\n",
      "[M::mem_process_seqs] Processed 22130 reads in 19.892 CPU sec, 20.014 real sec\n",
      "[M::process] read 21550 sequences (10000363 bp)...\n",
      "[M::mem_process_seqs] Processed 22202 reads in 21.361 CPU sec, 21.519 real sec\n",
      "[M::process] read 21248 sequences (10000733 bp)...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 21550 reads in 22.655 CPU sec, 22.803 real sec\n",
      "[M::process] read 1257 sequences (585731 bp)...\n",
      "[M::mem_process_seqs] Processed 21248 reads in 19.917 CPU sec, 20.103 real sec\n",
      "[M::mem_process_seqs] Processed 1257 reads in 1.227 CPU sec, 1.188 real sec\n",
      "[main] Version: 0.7.16a-r1181\n",
      "[main] CMD: /Users/jd/bwa-0.7.16a/bwa mem genome_subset.fasta barcode03.fa\n",
      "[main] Real time: 1478.548 sec; CPU: 1449.973 sec\n"
     ]
    }
   ],
   "source": [
    "#map new transcriptome sequences to draft genome\n",
    "!/Users/jd/bwa-0.7.16a/bwa mem genome_subset.fasta barcode03.fa > bar03_map.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools view -Sb bar03_map.sam > bar03_map.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 3 files...\r\n"
     ]
    }
   ],
   "source": [
    "!samtools sort bar03_map.bam -o bar03_map.sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!samtools index bar03_map.sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
